process-document/src/test/kotlin/com/ritense/processdocument/domain/impl/delegate/DocumentDelegateServiceTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.processdocument.domain.impl.delegate
+
+import com.ritense.document.domain.impl.JsonSchemaDocument
+import com.ritense.document.domain.impl.JsonSchemaDocumentId
+import com.ritense.document.service.DocumentService
+import com.ritense.document.service.impl.JsonSchemaDocumentService
+import com.ritense.processdocument.domain.impl.CamundaProcessInstanceId
+import com.ritense.processdocument.service.DocumentDelegateService
+import com.ritense.processdocument.service.ProcessDocumentService
+import org.camunda.community.mockito.delegate.DelegateExecutionFake
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import java.util.*
+
+internal class DocumentDelegateServiceTest {
+
+    lateinit var processDocumentService: ProcessDocumentService
+    lateinit var documentService: DocumentService
+    lateinit var jsonSchemaDocumentService: JsonSchemaDocumentService
+    lateinit var documentDelegateService: DocumentDelegateService
+
+    @BeforeEach
+    fun beforeEach() {
+        processDocumentService = mock()
+        documentService = mock()
+        jsonSchemaDocumentService = mock()
+        documentDelegateService = DocumentDelegateService(
+            processDocumentService,
+            documentService,
+            jsonSchemaDocumentService
+        )
+    }
+
+    @Test
+    fun `get id from document`() {
```

Comment:
Can you also write a test for the non-happy path(s)? Additionally, I would like to see tests for the other methods in the class as well.